### ISSUE TITLE
Support a container registry other than Docker Hub

### ISSUE THREAD

**Issue by gabe565** [TandoorRecipes/recipes#2375]
### Is your feature request related to a problem? Please describe.

Hello!

Yesterday, Docker sent an email stating they will be removing open source organizations within the next month, described by [this unofficial blog post](https://blog.alexellis.io/docker-is-deleting-open-source-images/).

Since then, Docker has apologized for the miscommunication in [this official blog post](https://www.docker.com/blog/we-apologize-we-did-a-terrible-job-announcing-the-end-of-docker-free-teams/), but I still feel that Docker Hub may not be a safe provider of open source container images anymore.

### Describe the solution you'd like

Push your images to an additional registry like [GHCR](https://ghcr.io) or [Quay](https://quay.io).

**Comment by vabene1111** [comment]
this should probably not be to hard to integrate into the existing action flow. Will need to look into this or someone is welcome to make a PR to the action files with the required code.

**Comment by gabe565** [comment]
Sounds good! I'll work on updating the CI. Would you mind if I consolidate some of these workflows while I'm at it? I could keep them separate, but I'd like to cut down on the number of duplicated workflows.

**Comment by vabene1111** [comment]
thanks for helping out with this.

**Comment by gabe565** [comment]
You're welcome! I'll make sure to keep the current tags, and allow the Raspi build to fail.

I noticed the Docker build action you're currently using doesn't support multiple repositories, so I would have to duplicate the build portion. Instead, I decided to swap to the [docker/build-push-action](https://github.com/docker/build-push-action) which supports multiple repositories and build cache.

I'm currently testing all of the workflows and edge cases in a fork, then I'll open a PR!
